src/RA_Core.cpp

```diff
@@ -1136,9 +1125,16 @@ std::string GetFolderFromDialog()
 {
     BROWSEINFO bi{};
     bi.hwndOwner = ::GetActiveWindow();
-
+#if _MBCS
     std::array<TCHAR, RA_MAX_PATH> pDisplayName{};
     bi.pszDisplayName = pDisplayName.data();
+#elif _UNICODE
+    auto pDisplayName = std::make_unique<TCHAR[]>(RA_MAX_PATH);
+    bi.pszDisplayName = pDisplayName.get();
+#else
+#error Unknown Character Set!
+#endif /* _MBCS */
```

Comment:
isn't the whole point of TCHAR to avoid things like this?

Reply:
Ra_max_path has a different value for unicode, 32,767 characters, which is too big for the stack.

Reply:
It has nothing to do with TCHAR